docs/remote/devcontainerjson-reference.md

```diff
@@ -104,6 +106,20 @@ Variables can be referenced in certain string values in `devcontainer.json` in t
 | `${localWorkspaceFolderBasename}` | Any | Name of the local folder that was opened in VS Code (that contains `.devcontainer/devcontainer.json`).<br /><br />⚠️ Not yet supported when using Clone Repository in Container Volume. |
 | `${containerWorkspaceFolderBasename}` | Any | Name of the folder where the workspace files can be found in the container. |
 
+## Lifecycle scripts
+
+When creating and working with a dev container, you may need different commands to be run at different points in the container's lifecycle. The below are properties that shape how the container is created and takes shape over time, and the order in which they are run (i.e. `onCreateCommand` will run after `initializeCommand`).
+
+| Property | Description |
+|----------|-------------|
+| `initializeCommand` | A command to run locally before anything else. If this is a single string, it will be run in a shell. If this is an array of strings, it will be run as a single command without shell. |
+| `onCreateCommand` | A command to run when creating the container. If this is a single string, it will be run in a shell. If this is an array of strings, it will be run as a single command without shell. <br> You may want to have some script steps happen before VS Code attaches (`onCreateCommand`) and some after (`postCreateCommand`). But, you can change this with `waitFor` to include even waiting for `postStartCommand` if you have something you want to happen every time. |
```

Comment:
- 'before/after VS Code attaches' is somewhat misleading. I think the difficulty in explaining this is that we have split up `postCreateCommand` in a way that only makes sense when you have the prebuild feature.
- `waitFor` doesn't change if/when the commands are run, it only affects at what time the VS Code UI may connect to the dev container. By default it can connect after `updateContentCommand`. I'd keep the documentation for `waitFor` in its own row/paragraph.

Reply:
Yeah see my other comment... I think we can say that by default we `waitFor` the `onCreateCommand`, but not `postCreateCommand`, but this can be changed using the `waitFor` property.

Reply:
Thanks for reviewing!

I removed the sentence about "You may want to have some script steps..." as it may not be entirely clear given the extent of the context here. I also moved `waitFor` to its own section below the table, rather than as an entry in the table, and I mentioned users may have VS Code `waitFor` the `onCreateCommand`. 

@chrmarti @Chuxel Please let me know if these changes seem accurate and logical to you.